eel_hole/utils.py

```diff
@@ -21,11 +46,124 @@ def rst_to_html(rst: str) -> str:
     )["html_body"]
 
 
-def clean_descriptions(datapackage: Package) -> Package:
-    if datapackage.description:
-        datapackage.description = rst_to_html(datapackage.description)
-    for resource in datapackage.resources:
-        resource.description = rst_to_html(resource.description)
-        for field in resource.schema.fields:
-            field.description = rst_to_html(field.description)
-    return datapackage
+def plaintext_to_html(text: str) -> str:
+    """
+    Convert plaintext with simple Markdown-style formatting to basic HTML.
+
+    This function escapes HTML characters and supports limited formatting:
+    - `**bold**` → <strong>bold</strong>
+    - `*italic*` → <em>italic</em>
+    - Unordered lists using `-` or `*` are wrapped in <ul>/<li> tags
+    - Ordered lists using `1.`, `2.`, etc. are wrapped in <ol>/<li> tags
+    - Double line breaks are interpreted as separating paragraphs
+    - Single line breaks are preserved as literal '\n' inside paragraphs (no <br> tags)
+
+    Args:
+        text (str): Plaintext input string with optional Markdown-style formatting.
+
+    Returns:
+        str: HTML-formatted string.
+    """
+    escaped = html.escape(text)
+
+    # Handle bold and italics (simple Markdown-style)
+    escaped = re.sub(r"\*\*(.+?)\*\*", r"<strong>\1</strong>", escaped)
+    escaped = re.sub(r"\*(.+?)\*", r"<em>\1</em>", escaped)
+
+    paragraphs = re.split(r"\n\s*\n", escaped)
+    html_parts = []
+
+    for para in paragraphs:
```

Comment:
Oh wow. This is a whole state machine. 🏆 for whoever wrote it, and the unit tests are adequate so I 100% believe it works, but is there really no drop-in markdown-lite widget we could've used instead?